Add initial codebase for TestNG repository (fixes #1)

lmalakhova opened the pull request.

lmalakhova committed:
Added the following:
- created a BaseTest class with corresponding before/after annotations
-created several test classes with tests and before/after configuration methods
-added TestNG dependency

sskorol commented on .idea/gradle.xml:
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
seems like you've pushed files from `.idea` folder... try to fix that with `rebase + edit` operation

sskorol commented on automation-testNG-examples.iml:
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
.iml files shouldn't be pushed as well

sskorol commented on build.gradle:
```diff
@@ -0,0 +1,20 @@
+dependencies {
+    compile('org.testng:testng:6.12')
+}
+task wrapper(type: Wrapper) {
```
add new lines between tasks

sskorol commented on build.gradle:
```diff
@@ -0,0 +1,20 @@
+}
+test {
+    useTestNG() { suites 'src/test/resources/smoke-suite.xml' }
+}
```
missing new line at the end of the file... should you add it to all you java and config files...

sskorol commented on src/test/java/com/waverley/AuthorizationTest.java:
```diff
@@ -0,0 +1,22 @@
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+public class AuthorizationTest extends com.waverley.BaseTest {
```
you don't need to specify a full path to `BaseTest`... use just a name...

sskorol commented on src/test/java/com/waverley/InitializationTest.java:
```diff
@@ -0,0 +1,18 @@
+    public void methodTwoTest() {
+        System.out.println("This is methodTwo");
+    }
+
```
redundant new line

sskorol reviewed (changes requested):
@lmalakhova fix comments please

lmalakhova committed:
Added the following:
- created a BaseTest class with corresponding before/after annotations
- created several test classes with tests and before/after configuration methods
- added TestNG dependency
- сomments have been fixed

lmalakhova committed:
Added the following:
- Add gradle quality plugin to the project.
- Fix all violations.
- Play with IInvokedMethodListener TestNG listener.
- Add test-data-supplier library (+ IntelliJ plugin).

sskorol commented:
@lmalakhova please check left comments... you still haven't removed redundant configuration files from PR
